app/components/createContainer/createContainerController.js

```diff
@@ -235,8 +231,8 @@ function ($scope, $state, $stateParams, Config, Info, Container, ContainerHelper
   }
 
   function prepareNetworkConfig(config) {
-    var mode = $scope.network.Mode;
-    var container = $scope.network.Container;
+    var mode = config.HostConfig.NetworkMode;
+    var container = $scope.formValues.NetworkContainer;
     var containerName = container;
     if (container && typeof container === 'object') {
       containerName = container.Names[0];
```

Comment:
Not working with Docker Swarm (see my comment below).

A condition should be appended here, if using swarm then the already defined filter `swarmcontainername` should be used.

See https://github.com/portainer/portainer/blob/develop/app/components/container/containerController.js#L16 for an example of the usage of a filter in a controller.